src/main/kotlin/it/pagopa/wallet/domain/details/CardDetails.kt

```diff
@@ -12,4 +13,14 @@ data class CardDetails(
 ) : WalletDetails {
     override val type: WalletDetailsType
         get() = WalletDetailsType.CARDS
+
```

Comment:
After refactoring as per below comment change this class as:
```
data class CardDetails(
    val bin: Bin,
    val maskedPan: MaskedPan,
    val expiryDate: ExpiryDate,
    val brand: WalletCardDetailsDto.BrandEnum,
    val holder: CardHolderName
) : WalletDetails<CardDetails> {
```
This way toDocument method will be bind to a specific data class instance instead of accepting any interface implementor

Reply:
Fixed in cdb1ad8